stdlib/public/core/CompilerProtocols.swift

```diff
@@ -183,11 +183,6 @@ public func != <T: Equatable>(lhs: T, rhs: T) -> Bool
 // definition above. (Compiler-synthesized hashing is based on stored properties
 // rather than rawValue; the difference is subtle, but it can be fatal.)
 extension RawRepresentable where RawValue: Hashable, Self: Hashable {
-  @inlinable // trivial
```

Comment:
I'm not familiar enough with this so it is more a question... it is safe to remove this? I've seen other places where this definition is kept to ensure ABI compatibility with 5.0 e.g. [here](https://github.com/apple/swift/blob/9116af96db813a9a7f718968d9f26b671037e25e/stdlib/public/Darwin/Foundation/NSStringEncodings.swift#L56) which seems to make sense. 
cc @airspeedswift @lorentey

Reply:
You're right, we must not remove this definition. I believe removing it would indeed be an ABI breaking change, although one that may be difficult to trigger in practice. However, there is another issue -- for RawRepresentable types, deleting this definition would replace this definition of `hashValue` with a compiler-generated one that would work in terms of `hash(into:)`, which could change the meaning of some existing code.

The correct way to resolve this issue is to fix the compiler diagnostic. It is only supposed to be emitted when a type implements `hashValue` but has a synthesized `hash(into:)` definition -- which should never trigger in this case.